src/SourcePathsMapping/include/SourcePathsMapping/MappingManager.h

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef SOURCE_PATHS_MAPPING_MAPPING_MANAGER_H_
+#define SOURCE_PATHS_MAPPING_MAPPING_MANAGER_H_
+
+#include <optional>
+#include <vector>
+
+#include "SourcePathsMapping/Mapping.h"
+
+namespace orbit_source_paths_mapping {
+
+// Maintains a list of source paths mappings and takes care of persistence using QSettings.
+class MappingManager {
+ public:
+  explicit MappingManager();
+  [[nodiscard]] const std::vector<Mapping>& GetMappings() const { return mappings_; }
+
+  void SetMappings(std::vector<Mapping> mappings);
+  void AppendMapping(Mapping mapping);
+
+  [[nodiscard]] std::optional<std::filesystem::path> MapToFirstMatchingTarget(
+      const std::filesystem::path& source_path) const {
+    return orbit_source_paths_mapping::MapToFirstMatchingTarget(mappings_, source_path);
+  }
+
+  [[nodiscard]] std::optional<std::filesystem::path> MapToFirstExistingTarget(
+      const std::filesystem::path& source_path) const {
+    return orbit_source_paths_mapping::MapToFirstExistingTarget(mappings_, source_path);
+  }
+
+ private:
+  void SaveMappings();
+  void LoadMappings();
+
+  std::vector<Mapping> mappings_;
```

Comment:
Yes, I considered this, but decided against this. Qt's serialization framework is great if you need to serialize a type in many different places in the code, because then it just works seamlessly. But since we only need to serialize in one place, I think, it's not worth the effort, or even harmful. Because the "seamlessness" comes with a cost. You have to declare operators, metatypes, implement functions. So the whole implementation is scattered over multiple files, at least a header file and one or two cpp files. So in this case I think going with explicity leads to shorter and more concise code. WDYT?